dhcp: T2265: refactor DHCP class

thomas-mangin opened the pull request:
Breaks the code down to make it easier to understand, simplify API

thomas-mangin committed:
dhcp: T2265: v4/v6 conflict for lease file

The vyatta-dhclient-hook is not differentiating between v4 and v6
and attempting to use the same lease file for both protocol.

Therefore only dhcp v4 or v6 could be setup on an interface and
not both.

thomas-mangin committed:
dhcp: T2265: refactor DHCP class

Break the code between v4 and v6, remove need for getter/setter
as they are just exposing the underlying dict.

Move FixedDict from tunnel code and expose it to other part so
it can be used to prevent accidental change to the dhcp option if
no default exists already.

c-po commented on python/vyos/ifconfig/dhcp.py:
```diff
@@ -16,103 +16,55 @@
-        """
-        self._dhcp[6]['options'] = options
+class _DHCPv4 (_DHCP):
+    template = """\
```
Please migrate the template to data/templates folder

thomas-mangin replied:
Hi Christian, I agree it should be done but could you please merge as the current code does not load them from there neither so it is not a step backward.

c-po replied:
Can‘t as there are conflicting files.

thomas-mangin replied:
argh ! ok - will re,re,...base :-) and add the template at the same time.

c-po reviewed (changes requested):
Please migrate the templates to data/templates folder

thomas-mangin committed:
template: T2230: helper to generate templates

Currently the pattern is to import jinja2 and re-generate
the template in every file. Dimitriy reported a reduction
in performance (1s) when commiting.

This code provide an helper function which caches the
Environment and template renderer, and can generate
template from one line (instead of the few currently)

thomas-mangin committed:
dhcp: T2230: use external template files

thomas-mangin commented:
migration to templated added to the patchset.
